backend/src/modules/doubles-tournament/core/application/event/EndTournamentMatchWhenMatchHasEnded.ts

```diff
@@ -0,0 +1,21 @@
+import { EventHandler } from '../../../../../shared/core/application/event/EventHandler';
+import { MatchHasEnded } from '../../../../match-module/core/domain/event/MatchHasEnded';
+import { DomainEventPublisher } from '../../../../../shared/core/application/event/DomainEventBus';
+import { CurrentTimeProvider } from '../../../../../shared/core/CurrentTimeProvider';
+import { TournamentMatchWasEnded } from '../../domain/event/TournamentMatchWasEnded';
+
+export class EndTournamentMatchWhenMatchHasEnded implements EventHandler<MatchHasEnded> {
+  constructor(private readonly eventPublisher: DomainEventPublisher, private readonly currentTimeProvider: CurrentTimeProvider) {}
+
+  async handle(event: MatchHasEnded): Promise<void> {
+    const [tournamentId, matchNumber] = event.matchId.split('_');
```

Comment:
Tutaj proponowalbym zroic cos takiego:
`const matchId = MatchId.fromRaw(event.matchId)`
I wtedy wyciagnac z tego to co zrobiłaś. Bo tak to ta logika w jaki sposób składamy MatchId jest w dwóch miejscach. Jesli ktos zmieni dzialanie klasy MatchId (np laczenie plusem), moze zapomniec o tym miejscu i nam nie bedzie dzialac.

Reply:
Tego nie rozumiem. Coś takiego...?
```
    const [eventTournamentId, eventMatchNumber] = event.matchId.split('_');
    const matchId = MatchId.from(eventTournamentId, Number(eventMatchNumber));
    const [tournamentId, matchNumber] = matchId.raw.split('_');
```

Reply:
No wlasnie fajnie, jakby to w MatchId siedziala wiedza jak ono jest kontstruowane. 
Czyli np:
```
    const matchId = MatchId.fromRaw(event.matchId);
    const tournamentId = matchId.tournamentId;
    const matchNumber = matchId.matchNumber
```
Wtedy widzisz, że zniknęło nam to split z zewnątrz i zostało tylko w klasie MatchId. Nikt nie wie w jaki sposóļ to jest złączane / rozdzielane - tylko ta klasa. Zmiana tego i wiedza o tym jest tylko tam.

Reply:
Aaaaaaa, ok :)

Reply:
https://github.com/nowakprojects/CodersCamp2020.Project.FullStack-Node-React.TableSoccerTournaments/pull/152/commits/64848fc094e595086d188b0073501866897f9dfb
Coś takiego?

Reply:
No akurat myslalem o utworzeniu obiektu MatchId i dodaniu do niego getterów. Bo teraz to troche dziwnie - mamy nazwane jak factory function to fromRaw ale w sumie nią nie jest, bo nie zwraca nam instancji klasy w której jest.